includes/reader-activation/class-reader-data.php

```diff
@@ -265,7 +265,32 @@ public static function api_delete_item( $request ) {
 	 * @param array $data      Data.
 	 */
 	public static function set_is_newsletter_subscriber( $timestamp, $data ) {
-		self::update_item( $data['user_id'] ?? \get_current_user_id(), 'is_newsletter_subscriber', true );
+		self::update_item( $data['user_id'], 'is_newsletter_subscriber', true );
```

Comment:
I still encounter the issue described here: https://github.com/Automattic/newspack-plugin/pull/2658#discussion_r1338714577 – after unsubscribing from all lists on the My Account page, the value of `*_is_newsletter_subscriber` in `localStorage` is `"true"`. This is because the value is set to true here, regardless of the list update outcome. If the user ends up with no lists they are subscribed to, `'is_newsletter_subscriber'` should be updated. 

A sidenote – `is_newsletter_subscriber` and `newsletter_subscribed_lists` seem redundant. The former can be surmised from the latter.

Reply:
I like the idea of deprecating the bool in favor of checking against the list. Implemented in 2fdde856087dd8e15a2d03f683e5ced81ea5beb6 and https://github.com/Automattic/newspack-popups/pull/1218/commits/ef849734e9436c1143d5aa2d8ff5832ffc435856

Reply:
Even though the `delete_item` deletes it from the database, the localStorage item will remain as we don't have an isomorphic deprecation strategy. I wonder if we should 🤔

Reply:
Decided to add support for deletion from server -> local in eefdbb99c948852579bd2a65600f815f0eef45cd

Any change in the subscription list selection should trigger the deletion of the bool item from the server, which also deletes from localStorage.

The deprecation will not occur on the `reader_logged_in` handler because that's async. It won't send the deleted item key back to the browser.

Reply:
I still see references to `is_newsletter_subscriber` in a few places ([1](https://github.com/Automattic/newspack-popups/blob/5b794adc26f808fca59ddc568cebda71b4440aa0/includes/class-newspack-segments-migration.php#L395), [2](https://github.com/Automattic/newspack-popups/blob/5b794adc26f808fca59ddc568cebda71b4440aa0/includes/class-newspack-segments-migration.php#L401), [3](https://github.com/Automattic/newspack-popups/blob/ada4246e7677ecb03213b40a32a4da8942f6a9be/src/view/README.md#L17)). Shouldn't these be updated?